File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/Response/ApplepayEchoResponse.cs
using System.Text;

namespace CsobGatewayClientExample.Communication.DataObjects.Response;

public class ApplepayEchoResponse : SignBase
{
    public int ResultCode { get; set; }
    public string? ResultMessage { get; set; }

    public ApplepayInitParams? InitParams { get; set; }

    public override string ToSign()
    {
        var sb = new StringBuilder();
        Add(sb, Dttm);
        Add(sb, ResultCode);
        Add(sb, ResultMessage);
        if (null != InitParams) Add(sb, InitParams.ToSign());
        DeleteLast(sb);
        return sb.ToString();
    }
}

File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/responses/ApplepayEchoResponse.java
package cz.monetplus.mips.eapi.v19.connector.entity.responses;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import cz.monetplus.mips.eapi.v19.connector.entity.SignBase;
import lombok.*;

@EqualsAndHashCode(callSuper = false)
@AllArgsConstructor @NoArgsConstructor
@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class ApplepayEchoResponse extends SignBase {
    private Integer resultCode;
    private String resultMessage;
    private ApplepayInitParams initParams;

    @Override
    public String toSign() {
        StringBuilder sb = new StringBuilder();
        add(sb, getDttm());
        add(sb, getResultCode());
        add(sb, getResultMessage());
        if(null != initParams) add(sb, initParams.toSign());
        deleteLast(sb);
        return sb.toString();
    }
}
